File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    if a == b or a % b == 0:
        answer = 1
    else:
        n = min(a,b)
        m = max(a,b)
        div = []
        for i in range(1,n+1):
            if n % i == 0:
                div.insert(0,i)
        for i in div:
            if m % i == 0:
                gcd = i
                break
        b = b //gcd
        answer = 1
        d = 2
        if b == 2 or b == 5:
            answer = 1
        else:
            while b > 1 and d <= b:
                if b % d == 0:
                    if d != 2 and d != 5:
                        answer = 2
                        break
                    else:
                        b //=d
                else:
                     d += 1   
    return answer

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        int answer = 1;
        if (a == b || a % b == 0) {
            answer = 1;
            return answer;
        }
        int m = Math.min(a, b);
        int M = Math.max(a, b);
        int gcd = 1;
        for (int i = m; i > 0; i--) {
            if (m % i == 0) {
                if (M % i == 0) {
                    gcd = i;
                    break;
                }
            }
        }
        b /= gcd;
        int d = 2;
        if (b == 2 || b == 5) {
            answer = 1;
        } else {
            while (b > 1 && d <= b) {
                if (b % d == 0) {
                    if (d != 2 && d != 5) {
                        answer = 2;
                        break;
                    } else {
                        b = b / d;
                    }
                } else {
                    d++;
                }
            }
        }
        return answer;
    }
}